src/games/strategy/triplea/ui/MapData.java

```diff
@@ -700,6 +701,50 @@ private Rectangle getBoundingRectWithTranslate(final List<Polygon> polys, final
   }
 
   /**
+   * Find the best rectangle inside the territory to place the name in. Finds the rectangle
+   * that can fit the name, that is the closest to the vertical center, and has a large width at
+   * that location. If there isn't any rectangles that can fit the name then default back to the
+   * bounding rectangle.
+   */
+  public Rectangle getBestTerritoryNameRect(final Territory t, final FontMetrics fm) {
+    final Rectangle territoryBounds = getBoundingRect(t);
+    Rectangle maxRectangle = territoryBounds;
+    final int minX = territoryBounds.x;
+    final int minY = territoryBounds.y;
+    final int maxX = minX + territoryBounds.width;
+    final int maxY = minY + territoryBounds.height;
+    final int centerY = minY + territoryBounds.height / 2;
+    final List<Polygon> polygons = m_polys.get(t.getName());
```

Comment:
Inlined. Not looking to change variable types outside this method especially since its not moved out of this class.